tutor/src/screens/assignment-edit/ux.js

```diff
@@ -47,8 +50,8 @@ export default class AssignmentUX {
       if (!course.pastTaskPlans.api.hasBeenFetched) {
         await course.pastTaskPlans.fetch();
       }
-      this.sourcePlanId = id;
-      this.plan = course.pastTaskPlans.get(id).createClone({ course });
+      this.plan = course.pastTaskPlans.get(id).createClone({ course, cloned_from_id: id });
+      this.isCloneOldAssignment = Boolean(course.pastTaskPlans.get(id).grading_template_id);
     } else {
```

Comment:
will pastTaskPlans always have a plan for the id?  If not the `get` will return undefined and it'll cause errors.  Maybe need to use lodash's `get`:
`Boolean(get(course.pastTaskPlans.get(id), 'grading_template_id', false));`

Reply:
Shouldn't there be always a plan from the previous course? Unless the instructor deletes the plan/assignment from the previous source? Can they do that?